### PR TITLE
Calculate admintools tag

### PR DESCRIPTION
**Pull request by tdeebswihart**
## What was changed

Our "release admintools" action now calculates the tag for you

### CODE REVIEW

File: .github/workflows/release-admin-tools.yml [hunk 1]
```diff
@@ -46,15 +47,32 @@ jobs:
...
       - name: Calculate admintools tag
         run: |
+          get_tag() {
+            # We need to remove the `shallow` marker for this submodule or else
+            # `git describe --tags` won't find our tags
+            cd "target/$1"
+            # We need to fetch the tags before git describe will do what we want
+            # We _only_ fetch the tags to save time
+            git fetch origin 'refs/tags/*:refs/tags/*' >&2
+            rm -f ../.git/modules/$1/shallow
```
Comment (dnr) [inline .github/workflows/release-admin-tools.yml:57]: we don't want to remove the `shallow` before the fetch? I'm not sure what it does exactly, just guessing

File: .github/workflows/release-admin-tools.yml [hunk 2]
```diff
@@ -46,15 +47,32 @@ jobs:
...
-          tctl_ver=$(cd target/tctl && git describe --tags | cut -d '-' -f-2)
-          echo "TAG=${today}${patch}+server-${temporal_ver}.cli-${cli_ver}.tctl-${tctl_ver}" >> "${GITHUB_ENV}"
+
+          tag="${today}${patch}+"
+          while read module; do
+            tag="${tag}${module}-$(get_tag ${module})."
+          done < <(git submodule status | cut -d ' ' -f3 | grep -v dockerize)
+
+          # remove trailing period
+          tag=$(echo "${tag}" | sed 's/\.$//')
```
Comment (dnr) [inline .github/workflows/release-admin-tools.yml:73]: we can use bash stuff here, right?
```suggestion
          tag="${tag%.}"
```

File: .github/workflows/release-admin-tools.yml [hunk 3]
```diff
@@ -46,15 +47,32 @@ jobs:
...
           patch=
           if [ -n "${{inputs.patch}}" ]; then
             patch="-p${{inputs.patch}}"
           fi
-          temporal_ver=$(cd target/temporal && git describe --tags | cut -d '-' -f-2)
-          cli_ver=$(cd target/cli && git describe --tags | cut -d '-' -f-2)
-          tctl_ver=$(cd target/tctl && git describe --tags | cut -d '-' -f-2)
-          echo "TAG=${today}${patch}+server-${temporal_ver}.cli-${cli_ver}.tctl-${tctl_ver}" >> "${GITHUB_ENV}"
+
+          tag="${today}${patch}+"
```
Comment (dnr) [inline .github/workflows/release-admin-tools.yml]: maybe slightly clearer:
```
tag=$(date "+%Y.%-m.%-d")
if [[ -n "${{inputs.patch}}" ]]; then
  tag="${tag}-p${{inputs.patch}}"
fi
tag="${tag}+"
```

File: .github/workflows/release-admin-tools.yml (hunk 2 above)
Comment (tdeebswihart) [reply .github/workflows/release-admin-tools.yml:73]: Oh right I can. Thanks

File: .github/workflows/release-admin-tools.yml (hunk 3 above)
Comment (tdeebswihart) [reply .github/workflows/release-admin-tools.yml]: Check the most recent version. We now loop over the submodules

**Review by dnr — APPROVED** [review]

**Pull request merged**